### PR TITLE
Add CI pipeline for linting and tests

### PR DESCRIPTION
**Pull request by m-juckes-pins**
Add CI pipeline

## Describe your changes

[DEVOPS BOARD DEV 61](https://pins-ds.atlassian.net/jira/software/projects/DEV/boards/205?selectedIssue=DEV-61)

Add CI Pipeline for linting and tests and PRs. Remove commit-lint from build and trigger only on merge

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: azure-pipelines-ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+              pathFilters:
+                - name: api
+                  path: packages/applications-service-api
+                - name: common
+                  path: packages/common
+                - name: web
+                  path: packages/forms-web-app
+                - name: back-office-subscribers
+                  path: packages/back-office-subscribers
+              stepName: check_files
```
Comment (beejjacobs) [inline azure-pipelines-ci.yml:56]: I don't think we need this, I think it was used by the build pipeline to check what to build.

**Review by beejjacobs — APPROVED** [review]
LGTM! Good work; one bit we might be able to remove, but no harm.

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**